stlab/concurrency/future.hpp

```diff
@@ -1739,6 +1719,10 @@ struct reduction_helper<future<void>> {
     future<void> value;
 };
 
+static inline auto reduce(future<future<void>>&& r) -> future<void> {
```

Comment:
Remove static and move this definition above - if this is circular we can use a tagged dispatch in a common - reduce to resolve or another technique.

Reply:
Agree, I will propose a fix in the next patch set.